contracts/lib/GenesisUtils.sol

```diff
@@ -83,25 +93,24 @@ library GenesisUtils {
      * @dev isGenesisState
      */
     function isGenesisState(uint256 id, uint256 idState) internal pure returns (bool) {
-        uint256 userSwappedState = reverse(idState);
-
-        bytes memory userStateB1 = int256ToBytes(userSwappedState);
+        bytes memory userStateB1 = int256ToBytes(idState);
 
         bytes memory cutState = BytesLib.slice(userStateB1, userStateB1.length - 27, 27);
 
-        bytes memory typDefault = hex"0000";
+        bytes memory userIdB = int256ToBytes(id);
+        bytes memory idType = BytesLib.slice(userIdB, userIdB.length - 31, 2);
 
-        bytes memory beforeChecksum = BytesLib.concat(typDefault, cutState);
+        bytes memory beforeChecksum = BytesLib.concat(idType, cutState);
         require(beforeChecksum.length == 29, "Checksum requires 29 length array");
 
-        uint16 s = sum(beforeChecksum);
+        uint16 checksum = reverse16(sum(beforeChecksum));
 
-        bytes memory checkSumBytes = abi.encodePacked(s);
+        bytes memory checkSumBytes = abi.encodePacked(checksum);
 
         bytes memory idBytes = BytesLib.concat(beforeChecksum, checkSumBytes);
         require(idBytes.length == 31, "idBytes requires 31 length array");
 
-        return id == reverse(toUint256(idBytes));
+        return id == uint248(bytes31(idBytes));
```

Comment:
yeah, it should be uint256

Reply:
It should be multiplable by 8. Than's why 248. Then I could convert to 256.
I got this approach from [here](https://ethereum.stackexchange.com/questions/51229/how-to-convert-bytes-to-uint-in-solidity#:~:text=uint256%20time%20%3D%20uint160(bytes20(0x2E4e72EDC83053F8ADE4a525191Ba7aBA086c067))%3B) - since we know the exect byte length, for me it looks more readable. 
I could try another approach if this one doesn't suit for us.